Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -1035,6 +1034,44 @@ AccessibilityObject* AccessibilityRenderObject::titleUIElement() const
 {
     if (m_renderer && isFieldset())
         return axObjectCache()->getOrCreate(dynamicDowncast<RenderBlock>(*m_renderer)->findFieldsetLegend(RenderBlock::FieldsetIncludeFloatingOrOutOfFlow));
+
+    if (is<RenderTableCell>(m_renderer.get())) {
+        // Try to find if the first cell in this row is a <th>. If it is,
+        // then it can act as the title ui element. (This is only in the
+        // case when the table is not appearing as an AXTable.)
+        if (isExposedTableCell())
+            return nullptr;
+
+        // Table cells that are th cannot have title ui elements, since by definition
+        // they are title ui elements
+        if (WebCore::elementName(checkedNode().get()) == ElementName::HTML_th)
+            return nullptr;
+
+        CheckedRef renderCell = downcast<RenderTableCell>(*m_renderer);
+
+        // If this cell is in the first column, there is no need to continue.
+        int col = renderCell->col();
+        if (!col)
+            return nullptr;
+
+        int row = renderCell->rowIndex();
```

Comment:
I know you didn't write this, but should we move this to after the `CheckedPtr section`, closer to where it's actually used?